controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java

```diff
@@ -0,0 +1,93 @@
+package com.yahoo.vespa.hosted.controller.deployment;
+
+import com.yahoo.config.provision.ApplicationId;
+import com.yahoo.vespa.hosted.controller.Controller;
+import com.yahoo.vespa.hosted.controller.api.integration.LogStore;
+import com.yahoo.vespa.hosted.controller.api.integration.zone.ZoneId;
+
+import java.util.List;
+
+/**
+ * A singleton owned by the controller, which contains the state and methods for controlling deployment jobs.
+ *
+ * Keys are the {@link ApplicationId} of the real application, for which the deployment job is run, and the
+ * {@link ZoneId} of the real deployment to test.
+ *
+ * Although the deployment jobs are themselves applications, their IDs are not to be referenced.
+ *
+ * @author jonmv
+ */
+public class JobController {
+
+    private final Controller controller;
+    private final LogStore logs;
+
+    public JobController(Controller controller) {
+        this.controller = controller;
+        this.logs = null;
+    }
+
+
+// GET:
+    /** Returns whether the given application has registered with this build service. */
+    boolean builds(ApplicationId application) {
+        return false;
+    }
+
+    /** Returns a list of all application which have registered. */
+    List<ApplicationId> applications() {
```

Comment:
Why do we need the registering thing? Seems like double accounting.

Reply:
What's the duplicate? We need to know, somehow, whether a given application uses this build service, or our default one. You could infer that from whether it has any test-jars stored in the artefact store, but ... no. 
It could also be stored in the `Application`, I suppose, but I didn't find that very tidy.